pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1071,12 +1071,20 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.ClientTLSS
 		if c.Http2ProtocolOptions != nil {
 			// This is HTTP/2 in-mesh cluster, advertise it with ALPN.
 			if tls.Mode == networking.ClientTLSSettings_ISTIO_MUTUAL {
-				tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNInMeshH2
+				// Enable sending 	istio-peer-exchange	 ALPN in ALPN list if TCP
```

Comment:
This comment isn't very useful, it says what is happening which is obvious from reading the code, but doesn't say *why* which is pretty hard to determine without digging back through this PR/issue

Reply:
Added as per suggestion from Mandar too